test/server/core/gameSteps/phases/SetupPhase.spec.ts

```diff
@@ -0,0 +1,185 @@
+                            'atst',
+                            'atst',
+                            'wampa',
+                            'atst',
+                            'atst',
+                        ],
+                    }
+                });
+            });
+
+            it('should flow normally through each step', function () {
+                const { context } = contextRef;
+
+                // The setup phase is divided into 3 or 4 steps: choose Initiative player, Draw card step, mulligan step (optional), resource step
+                // Choose Initiative Step
+                context.selectInitiativePlayer(context.player1);
+
+                // Draw cards step
+                expect(context.player1.handSize).toBe(6);
+                expect(context.player2.handSize).toBe(6);
+                const beforeMulliganHand = context.player1.hand;
+                const beforePlayer2Hand = context.player2.hand;
+
+                // Mulligan step
+                context.player1.clickPrompt('yes');
+                context.player2.clickPrompt('no');
+                const afterMulliganHand = context.player1.hand;
+                const afterPlayer2Hand = context.player2.hand;
+                expect(beforeMulliganHand).not.toEqual(afterMulliganHand);
+                expect(beforePlayer2Hand).toEqual(afterPlayer2Hand);
+
+                // Resource step
+                // check that no resource was automatically set
+                expect(context.player1.resources.length).toBe(0);
+                expect(context.player2.resources.length).toBe(0);
+
+                const player1FirstCard = context.player1.hand[0];
+                const player1SecondCard = context.player1.hand[1];
+
+                // We check if player1's hand has the only selectable cards
+                expect(context.player1).toBeAbleToSelectExactly(context.player1.hand);
+
+                // we check if player2's hand has the only selectable cards
+                expect(context.player2).toBeAbleToSelectExactly(context.player2.hand);
+
+                // select 2 cards to resource player1 and check prompts
+                context.player1.clickCard(context.player1.hand[0]);
+
+                // We check if player1's hand has the only selectable cards
+                expect(context.player1).toBeAbleToSelectExactly(context.player1.hand);
+                expect(context.player1).toHaveExactPromptButtons(['Done']);
+                expect(context.player2).toHaveExactPromptButtons(['Done']);
+
+                context.player1.clickCard(context.player1.hand[1]);
+                expect(context.player1).toHaveExactPromptButtons(['Done']);
+                expect(context.player2).toHaveExactPromptButtons(['Done']);
+
+                // Check that player1 cannot select any additional cards
+                context.player1.clickCardNonChecking(context.player1.hand[2]);
+                expect(context.player1.selectedCards.length).toBe(2);
```

Comment:
These tests are awesome!! Do you mind also adding similar checks of the card selection process to the resource step of the setup phase, to the degree it makes sense?

Reply:
do you mean the setupPhase or the regroup phase?

Reply:
Added some additional ones in the setupPhase for good measure and some for the regroup Phase.